library-viewbinding/src/main/java/com/xwray/groupie/viewbinding/BindableItem.java

```diff
@@ -0,0 +1,74 @@
+package com.xwray.groupie.viewbinding;
+
+import androidx.annotation.NonNull;
+import androidx.viewbinding.ViewBinding;
+
+import android.view.View;
+
+import com.xwray.groupie.Item;
+
+import java.util.List;
+
+/**
+ * The base unit of content for a GroupAdapter.
+ * <p>
+ * Because an Item is a Group of size one, you don't need to use Groups directly if you don't want;
+ * simply mix and match Items and add directly to the adapter.
+ * <p>
+ * If you want to use Groups, because Item extends Group, you can mix and match adding Items and
+ * other Groups directly to the adapter.
+ *
+ * @param <T> The ViewBinding subclass associated with this Item.
+ */
+public abstract class BindableItem<T extends ViewBinding> extends Item<GroupieViewHolder<T>> {
```

Comment:
I think it is OK that the names `BindableItem` and `GroupieViewHolder` is same as the corresponding classes in `library-databinding` because they have different package path. If Adding some prefix is preferable, I think it should be added to the classes in both modules.

Reply:
ViewBinding and DataBinding both are supposed to co-exist in a project.
There is a possibility of them being used in the same file (kotlin file with multiple Item class definition)
In such cases, one of the class usage will have to be fully qualified, because of the same name.
I also understand that renaming BindableItem is not a good idea for the existing users.

Reply:
Thanks for your review :)

As I [commented below](https://github.com/lisawray/groupie/pull/325#issuecomment-589940382), `BindableItem<T extends ViewBinding>` should work not only with ViewBinding classes but also with DataBinding classes. So if an app which enables both DataBinding and ViewBinding want to use groupie with its extension library, depending only on `library-viewbinding` satisfies its demand and collision of class name will not happen.

| binding library | extension to be used |
| --- | --- |
| DataBinding only | `groupie-databinding`
| ViewBinding only | `groupie-viewbinding` |
| Both DataBinding and ViewBinding | `groupie-viewbinding` |

Reply:
This can probably work the other way around.

binding library | extension to be used
-- | --
ViewBinding only | groupie-viewbinding
Both DataBinding and ViewBinding | groupie-databinding

We will need to update `BindableItem<T extends ViewDataBinding>` to `BindableItem<T extends ViewBinding>` in the groupie-databinding artifact.

But we still need to change the abstract method
```
@NonNull
    protected abstract T initializeViewBinding(@NonNull View view);
```
to
```
    @NonNull
    protected T initializeBinding(@NonNull View view) {
        return DataBindingUtil.bind(view)
    }
```
This should be overridden with custom implementation when using ViewBinding.

Reply:
> We will need to update `BindableItem<T extends ViewDataBinding>` to `BindableItem<T extends ViewBinding>` in the groupie-databinding artifact.

This requires the users of groupie-databinding to update Android Gradle Plugin >= 3.6.0 (alpha11) after the update of groupie including ViewBinding support, but I think this may be OK if announcing that the update includes breaking change 👀
In this case, we don't have to add another module, and what to do is to change groupie-databinding to support all of above use-case.

> But we still need to change the abstract method ...  This should be overridden with custom implementation when using ViewBinding.

This change cannot pass compile because `T` does not always extends `ViewDataBinding`, so the user should always override `initializeViewBinding` even if they use only DataBinding.

Reply:
> This change cannot pass compile because T does not always extends ViewDataBinding, so the user should always override initializeViewBinding even if they use only DataBinding.

In a project with hundreds of Item classes, It will be a nightmare.

Reply:
Yes, so I don't think adding the consideration to ViewBinding to existing `groupie-databinding` is good idea and I'd like to add new library (= `groupie-viewbinding` in this pull request) for the support to both ViewBinding and DataBinding because new library with new feature will not cause breaking change for users of the existing library.
If you are also concerned with the migration from DataBinding-only to both DataBinding and ViewBinding, and you don't want to change existing Item classes extending current `BindableItem`, then we have no other choice than to use reflection to get `bind` method from type parameter `T`. I want to avoid using reflection if possible.